### ISSUE TITLE
listConnections with parameter is not honoring parameter

### ISSUE THREAD

**Issue by zhawtof** [NangoHQ/nango#654]
**Setup:** Local development
**Environment:** NodeJS

tl;dr `nango.listConnections(<connection_id>)` is operating as if `<connection_id>` is null.

Here is my code:
```ts
console.log('UserID:', userId, typeof userId);
const connections = await nango.listConnections(userId);
console.log(connections);
```

Expected:
```
UserID:  b953f16f-5c24-4f3d-bdfc-6d2593ae53ff string
{
  connections: [
    {
      id: 11,
      connection_id: 'b953f16f-5c24-4f3d-bdfc-6d2593ae53ff',
      provider: 'salesforce',
      created: '2023-06-02T02:46:54.554Z'
    }
  ]
}
```

Actual:
```
UserID:  b953f16f-5c24-4f3d-bdfc-6d2593ae53ff string
{
  connections: [
    {
      id: 6,
      connection_id: '4ae14089-49be-46b1-b623-c60945d7d264',
      provider: 'salesforce',
      created: '2023-05-18T05:36:58.491Z'
    },
    {
      id: 11,
      connection_id: 'b953f16f-5c24-4f3d-bdfc-6d2593ae53ff',
      provider: 'salesforce',
      created: '2023-06-02T02:46:54.554Z'
    },
    {
      id: 4,
      connection_id: 'test-connection-id',
      provider: 'salesforce',
      created: '2023-05-18T03:34:25.909Z'
    },
    {
      id: 2,
      connection_id: '4ae14089-49be-46b1-b623-c60945d7d264',
      provider: 'zoom',
      created: '2023-05-17T06:08:40.018Z'
    },
    {
      id: 1,
      connection_id: 'test-connection-id',
      provider: 'zoom',
      created: '2023-05-17T05:54:41.058Z'
    }
  ]
}
```

You can see that it's operating as if `connectionId` was undefined.

---

## On a separate note
It's also worth mentioning that `listConnections` is written in a way to expect unexpected consequences. If `connectionId` is ever `undefined` unintentionally, then listConnections continues to operate and exposes connections well beyond what was intended. Most users would expect an error here rather than list every single possible connection.

I would recommend creating a `listConnectionsForId` function and deprecating the `listConnections` which takes a parameter.

**Comment by khaliqgant** [comment]
Thanks for reporting @zhawtof! Resolved with @nangohq/node@0.17.2 which is also tagged as latest.

Agree regarding the confusion of the usage of `listConnections` and the optional param. We'll do some cleanup of that function to make the usage clearer.
